fe/src/main/java/org/apache/doris/external/EsIndexState.java

```diff
@@ -76,66 +68,45 @@ public TNetworkAddress randomAddress(Map<String, EsNodeInfo> nodesInfo) {
         EsNodeInfo[] nodeInfos = (EsNodeInfo[]) nodesInfo.values().toArray();
         return nodeInfos[seed].getPublishAddress();
     }
-    
-    public static EsIndexState parseIndexStateV55(String indexName, JSONObject indicesRoutingMap, 
-            JSONObject nodesMap, 
-            JSONObject indicesMetaMap, PartitionInfo partitionInfo) throws AnalysisException {
+
+    public static EsIndexState parseIndexState(String indexName, JSONObject nodesMap,
+        JSONArray shards) {
         EsIndexState indexState = new EsIndexState(indexName);
-        JSONObject shardRoutings = indicesRoutingMap.getJSONObject(indexName).getJSONObject("shards");
-        for (String shardKey : shardRoutings.keySet()) {
+        int length = shards.length();
+        for (int i = 0; i < length; i++) {
             List<EsShardRouting> singleShardRouting = Lists.newArrayList();
-            JSONArray shardRouting = shardRoutings.getJSONArray(shardKey);
-            for (int i = 0; i < shardRouting.length(); ++i) {
-                JSONObject shard = shardRouting.getJSONObject(i);
+            JSONArray shardsArray = shards.getJSONArray(i);
+            int arrayLength = shardsArray.length();
+            for (int j = 0; j < arrayLength; j++) {
+                JSONObject shard = shardsArray.getJSONObject(j);
                 String shardState = shard.getString("state");
                 if ("STARTED".equalsIgnoreCase(shardState)) {
```

Comment:
take the `relocating` state into account maybe as well?

Reply:
Ok, I will modify it